src/mixins/java/org/spongepowered/common/mixin/api/minecraft/world/entity/monster/CreeperMixin_API.java

```diff
@@ -36,19 +36,21 @@
 public abstract class CreeperMixin_API extends MonsterMixin_API implements Creeper {
 
     // @formatter:off
-    @Shadow public void shadow$ignite() { } // explode
+    @Shadow protected abstract void shadow$explodeCreeper();
     // @formatter:on
 
+
     @Override
     public void detonate() {
-        this.shadow$ignite();
+        this.shadow$explodeCreeper();
```

Comment:
This is going to be a breaking change due to the behavior change. We would want to have another API method to allow igniting the creeper.

Reply:
Just to clarify, the original behavior wasn't implemented correctly based on the API. The `detonate` method should trigger an immediate explosion ("Instructs this explosive to detonate as soon as possible.") as seen with the TNT minecart (you can only make it explode, you cannot ignite it with the API). I will still revert this change.